src/odemis/gui/cont/streams.py

```diff
@@ -2990,3 +2992,356 @@ def _updateScanStage(self, use):
+        self.model.name.value = txt
+
+
+class FastEMCalibrationController(object):
+    """
+    Listens to the calibration buttons and creates the FastEMROCControllers accordingly.
+    """
+
+    def __init__(self, tab_data, calibration_bar, view_ctrl):
+        """
+        tab_data (FastEMAcquisitionGUIData): tab data model
+        calibration_bar (FastEMCalibrationBar): main calibration panel
+        view_ctrl (FastEMAcquisitionViewport): viewport controller
+        """
+        self._view_ctrl = view_ctrl
+        self._calibration_bar = calibration_bar
+        self._tab_data = tab_data
+
+        self.panel = FastEMCalibrationPanel(calibration_bar, tab_data.scintillator_layout)
+        calibration_bar.add_calibration_panel(self.panel)
+
+        self.roc_ctrls = {}  # int --> FastEMROCController
+
+        for btn in self.panel.buttons.values():
+            btn.Bind(wx.EVT_BUTTON, self._on_button)
+
+    def _on_button(self, evt):
+        btn = evt.GetEventObject()
+        btn.SetLabel("OK")
+        btn.SetForegroundColour(wx.GREEN)
+
+        num = [num for num, b in self.panel.buttons.items() if b == btn][0]
+        if num not in self.roc_ctrls:
+            # Create new calibration controller
+            logging.debug("Adding ROC controller %s.", num)
+            roc_ctrl = FastEMROCController(num, self._tab_data, self._view_ctrl)
+            self.roc_ctrls[num] = roc_ctrl
+        else:
+            # Zoom to existing calibration region
+            roc_ctrl = self.roc_ctrls[num]
+        roc_ctrl.zoom_in()
+
+
+class FastEMROCController(object):
+    """
+    Controller for a single region of calibration.
+    """
+
+    def __init__(self, number, tab_data, view_ctrl):
+        """
+        number (int): number of the calibration region
+        tab_data (FastEMAcquisitionGUIData): tab data model
+        view_ctrl (FastEMAcquisitionViewport): viewport controller
+        """
+        self._view_ctrl = view_ctrl
+        self._tab_data = tab_data
+
+        # By default, the calibration region is in the center of the scintillator. The size is given by
+        # the ebeam resolution.
+        pos = tab_data.scintillator_positions[number]
+        sz = (tab_data.main.ebeam.resolution.value[0] * tab_data.main.ebeam.pixelSize.value[0],
+              tab_data.main.ebeam.resolution.value[1] * tab_data.main.ebeam.pixelSize.value[1])
+
+        l = pos[0] - 0.5 * sz[0]
+        t = pos[1] + 0.5 * sz[1]
+        r = pos[0] + 0.5 * sz[0]
+        b = pos[1] - 0.5 * sz[1]
+
+        # Get ROC model (exists already in tab data) and change coordinates
+        self.model = tab_data.calibration_regions.value[number]
+        self.model.coordinates.value = (l, t, r, b)
+        self.model.coordinates.subscribe(self._on_coordinates)
+
+        # Add overlay
+        self.overlay = view_ctrl.viewports[0].canvas.add_calibration_overlay(self.model.coordinates, number)
+
+    def zoom_in(self):
```

Comment:
We have other similar behaving functions which are called "fitViewToContent". So I'd suggest to name it the same or something similar, like "fit_view_to_bbox".